pageserver/src/walredo.rs

```diff
@@ -204,6 +204,18 @@ impl PostgresRedoManager {
     pub(crate) fn maybe_quiesce(&self, idle_timeout: Duration) {
         if let Ok(g) = self.last_successful_redo_at.try_lock() {
             if let Some(last_successful_redo_at) = *g {
+                // Kill the walredo process if
+                // - it has been unused for `idle_timeout`
+                // - it has been used, but, without success.
+                // The former is just good housekeeping.
+                // The latter adds robustness for the case where something is wrong
+                // with the walredo process.
+                //
+                // Note that we don't want to kill the process immediately on each redo failure.
+                // The reason is that the redo failure could be caused by corrupted or malicious data.
```

Comment:
I have missed this one, checking...

Reply:
But we do kill it on each failure? https://github.com/neondatabase/neon/blob/fa52cd575e8bec9cd791f933ca80c498b17be7fa/pageserver/src/walredo.rs#L314-L327

Reply:
Hm, right, I had missed that :(

So, should we remove that eager killing?

Reply:
I think we should keep killing it, we have no guarantees it will continue to work after a failure because only way we will fail is by not receiving the 8kB in timely fashion.

If it's not giving out output, we are in a protocol busted situation and cannot recover. However... you just removed the timeout.